### PR TITLE
checks scaling limit in docs

### PR DESCRIPTION
**Pull request by johannkm**
document checks ui limits

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17769" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17769?utm_source=stack-comment).

File: docs/content/concepts/assets/asset-checks.mdx [hunk 1]
```diff
@@ -285,3 +285,9 @@ defs = Definitions(
     schedules=[basic_schedule],
 )
 ```
+
+---
+
+## Scaling limits
```
Comment (johannkm) [inline docs/content/concepts/assets/asset-checks.mdx:291]: `## Limitations`?

Should this be its own section? Could also add a section about defining multiple checks on one asset

Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-checks.mdx:291]: +1 to it being **Limitations**. It gives us room to add more if we need to, and this is consistent with other docs/features that also have limitation sections.

Also, I feel pretty strongly about having limitation sections near the top of pages versus the bottom. As a user, I'd prefer to know what the limits of things are upfront. I've been reworking concept pages to follow this structure:

```
- Title/overview
- Prerequisites (if any)
- Limitations (if any)
- Relevant APIs
- Defining [concept]
- ...
```

Open to thoughts/feedback on this!

Comment (sryza) [reply docs/content/concepts/assets/asset-checks.mdx:291]: I like the idea of calling it Limitations and being consistent with other docs.

I'm hesitant about putting it near the top because I think the limitations we're included here will be irrelevant information to the vast majority of users. You'd need to be using the feature in a very weird and unexpected way to actually hit this limitation.

Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-checks.mdx:291]: Other than the limitation with the UI, is there anything else that we know of that would be a limitation? Otherwise could we just embed this info in a callout in the Executing > UI section?

Comment (johannkm) [reply docs/content/concepts/assets/asset-checks.mdx:291]: It doesn't quite fit there since it's about how many checks you can define. We could call out that we don't have partition support as a limitation.

Comment (johannkm) [reply docs/content/concepts/assets/asset-checks.mdx]: updated

File: docs/content/concepts/assets/asset-checks.mdx [hunk 2]
```diff
@@ -285,3 +285,11 @@ defs = Definitions(
...
 )
 ```
+
+---
+
+## Limitations
+
+- Dagster's UI is tested with a maximum of 1000 checks per asset. It's designed with the expectation that most assets will have fewer than 50 checks. If you have a use case that doesn't fit these limits, you can reach out to discuss.
+
+- Checks are currently only supported per-asset, not per-partition. See [this issue](https://github.com/dagster-io/dagster/issues/17005) for more information.
```
Comment (erinkcochran87) [inline docs/content/concepts/assets/asset-checks.mdx:295]: This feels nitpicky (sorry!) but I'd rephrase the second sentence here - the linked issue doesn't really provide any additional information. Maybe something like "See this issue for updates" if we're going to use it for tracking implementation effort.

**Review by erinkcochran87 — APPROVED** [review]
Just one comment, but otherwise good with this. Stamping to unblock ya
